package/MDAnalysis/analysis/waterdynamics.py

```diff
@@ -521,79 +517,67 @@ def _getOneDeltaPoint(self, universe, repInd, i, t0, dt):
             valHH += self.lg2(np.dot(unitHHVector0, unitHHVectorp))
             valdip += self.lg2(np.dot(unitdipVector0, unitdipVectorp))
             n += 1
-        return  (valOH/n, valHH/n, valdip/n) if n > 0 else (0, 0, 0)
-
+        if n > 0:
+            return valOH/n, valHH/n, valdip/n
+        else:
+            return 0, 0, 0
 
-    def _getMeanOnePoint(self, universe, selection1, selection_str, dt,
-                         totalFrames):
+    def _get_mean_one_point(self, universe, selections):
         """
         This function gets one point of the plot C_vec vs t. It uses the
         _getOneDeltaPoint() function to calculate the average.
-
         """
-        repInd = self._repeatedIndex(selection1, dt, totalFrames)
-        sumsdt = 0
-        n = 0.0
-        sumDeltaOH = 0.0
-        sumDeltaHH = 0.0
-        sumDeltadip = 0.0
+        repeated_indices = self._get_repeated_indices(selections, self.frame)
+        cum_frames = 0
+        sumDeltaOH = 0
+        sumDeltaHH = 0
+        sumDeltadip = 0
+        n = 0
 
-        for j in range(totalFrames // dt - 1):
-            a = self._getOneDeltaPoint(universe, repInd, j, sumsdt, dt)
+        for i in range(self.start, self.stop // self.frame - 1):
```

Comment:
I don't understand why `self.stop // self.frame - 1`. Can you add a comment here explaining what this loop does?